### PR TITLE
Nested layout ui improvements (fixes #6)

### LINKED ISSUE #6: Nested forms visibility issues
In our project we want to render automatic forms for json-schemas that may have more than one nested object on it, ([for example](https://gist.github.com/pefernan/7dfe047edda43ef2896e11ccd6b1ad32)).

In this case our schema has the following properties:

- flight: object
- hotel: object
  - address: object (nested to hotel)

When rendering an autoform, it correctly generates the form with two nested fields for flight and hotel and inside of hotel another one for address. But there are some visibilty issues that may be addressed:

- There is some spacing missing inside the nested fields. This appears to be related to a div that wraps the nested form that kind of breaks the grid.
- Since the form shows all fields at same level, It's hard to see what fields belong to what object. In the attached screenshot it's possible to see that on the **Hotel** nested form, we have the **Address** nested form, and just after it there are other fields that belong the **Hotel** (Booking Number, Name, Phone & Room).

image

### PR DESCRIPTION
**Pull request by kingsleyzissou**
I have added a card layout to try and address the ui issues around nested fields.

Here is a screenshot of the form based on the schema provided by @pefernan

image

### CODE REVIEW

**Comment by wtrocki** [comment]
Can we put better header level on the nested ones. 

So for example top one gets h1, then you get h2,h3 for nested ones.
This means it will be easy to style it even without class names or ids

**Comment by wtrocki** [comment]
I mean `<h3>Hotel</h3></h4>Address</Address>`?

File: src/NestField.tsx [hunk 1]
```diff
@@ -25,16 +26,25 @@ const Nest = ({
...
     >
-      {label && <label>{label}</label>}
-      {children
-        ? injectName(name, children)
-        : fields?.map(key => (
-            <AutoField key={key} isDisabled={props.disabled} name={joinName(name, key)} {...itemProps} />
-          ))}
-    </div>
+      <CardBody>
+        {label && <label>{label}</label>}
```
Comment (wtrocki) [inline src/NestField.tsx:33]: Checking if ionic supports h4 etc. We can wrap them in label itself.

Comment (kingsleyzissou) [reply src/NestField.tsx:33]: I could look into it, but I'm not too sure how we would get the information about the depth of the object. i.e. we would probably have to traverse the whole object to get the level (we could try using `traverse-js` to do this). But I'm not sure about the cost-benefit for this.

Ionic is a bit weird too, so even if you put those tags in, the components default the typography to the Ionic component's default and so the `<h1>`, `<h2>` tags and so on don't have any effect.

Comment (wtrocki) [reply src/NestField.tsx:33]: ah.. that is the challenge. I will log that as issue and try it myself.

**Comment by wtrocki** [comment]
Works really good!

**Comment by wtrocki** [comment]
Rebase needed

**Comment by wtrocki** [comment]
Done rebase

**Pull request merged**